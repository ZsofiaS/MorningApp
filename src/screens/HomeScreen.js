import React from 'react';
import { StyleSheet, Text, View, Image} from 'react-native';
import { Button, ThemeProvider } from 'react-native-elements';
import { StatusBar } from 'expo-status-bar';
import { withNavigation } from 'react-navigation';
import colours from '../../constants/colours.js';
import OptionCard from '../components/OptionCard';

export default function HomeScreen({navigation}) {
  return (
      <View style={styles.container}>
        <Text style={styles.title}>Have a great morning!</Text>
        <View style={styles.listContainer}>
          <OptionCard
            source={require('../../images/lotus-position-2.png')}
            onPress={() => navigation.navigate('Breath')}
            title="Take a break and breath"
          />
          <OptionCard
            source={require('../../images/branch.png')}
            onPress={() => navigation.navigate('Gratitude')}
            title="Gratitude Journal"
          />
          <OptionCard
            source={require('../../images/lotus.png')}
            onPress={() => navigation.navigate('Affirmations')}
            title="Daily affirmation"
          />
        </View>
        <StatusBar
            style="auto" />
      </View>
    )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-start',

  },
  title: {
    color: colours.purple,
    fontSize: 25,
    marginVertical: 100,
    paddingHorizontal: 50
  },
  image: {
    marginVertical: 10,
    marginHorizontal: 10,
    width: 60,
    height: 60,
  },
  listContainer: {
    marginVertical: -15
  },
  bContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 10
  }
});
