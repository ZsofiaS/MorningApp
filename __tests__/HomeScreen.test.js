import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../src/screens/HomeScreen';

test('homescreen renders correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
})
