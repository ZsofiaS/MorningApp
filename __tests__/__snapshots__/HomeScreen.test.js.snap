// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homescreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#A167A5",
        "fontSize": 25,
        "marginVertical": 100,
        "paddingHorizontal": 50,
      }
    }
  >
    Have a great morning!
  </Text>
  <View
    style={
      Object {
        "marginVertical": -15,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 60,
            "marginHorizontal": 10,
            "marginVertical": 10,
            "width": 60,
          }
        }
      />
      <View
        style={
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#c48b9f",
                  "fontSize": 20,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Take a break and breath
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 60,
            "marginHorizontal": 10,
            "marginVertical": 10,
            "width": 60,
          }
        }
      />
      <View
        style={
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#c48b9f",
                  "fontSize": 20,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Gratitude
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 60,
            "marginHorizontal": 10,
            "marginVertical": 10,
            "width": 60,
          }
        }
      />
      <View
        style={
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#c48b9f",
                  "fontSize": 20,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Daily affirmation
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
